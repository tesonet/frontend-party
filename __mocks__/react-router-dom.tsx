import React, { ReactChildren } from "react";
import * as rrd from "react-router-dom";

const MockBrowserRouter: React.FC<{ children: ReactChildren }> = ({
  children
}) => <div>{children}</div>;

/* eslint-disable @typescript-eslint/no-explicit-any */
(rrd as any).BrowserRouter = MockBrowserRouter;

module.exports = rrd;
