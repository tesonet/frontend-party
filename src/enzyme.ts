import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import fetchMock from "jest-fetch-mock";

declare let global: { fetch: {} };

global.fetch = fetchMock;
configure({ adapter: new Adapter() });
