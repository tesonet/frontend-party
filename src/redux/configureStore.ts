import { createStore, compose, applyMiddleware } from 'redux';
import { RouterState, routerMiddleware } from 'connected-react-router';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { AuthState } from './containers/auth/authReducer';
import { ServersState } from './containers/servers/serversReducer';
import { createRootReducer, history } from './rootReducer';

export interface AppState {
  auth: AuthState;
  servers: ServersState;
  router: RouterState;
}

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['auth']
};

const persistedReducer = persistReducer(persistConfig, createRootReducer());

export const configureStore = () => {
  const store = createStore(
    persistedReducer,
    compose(applyMiddleware(routerMiddleware(history)))
  );

  const persistor = persistStore(store);

  if (module.hot) {
    module.hot.accept('./rootReducer'), () => {
      const nextRootReducer = require('./rootReducer').createRootReducer();
      store.replaceReducer(persistReducer(persistConfig, nextRootReducer));
    };
  }

  return { store, persistor, history };
}