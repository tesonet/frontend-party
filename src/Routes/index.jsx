import React from 'react';
import { Route } from 'react-router-dom';

import PrivateRoutes from './PrivateRoutes';
import Login from '../views/Login';

const Routes = () => (
  <div>
    <Route component={PrivateRoutes} />
    <Route exact path="/login" component={Login} />
  </div>
);

export default Routes;
