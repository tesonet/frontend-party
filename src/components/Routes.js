import React from "react";
import { HashRouter, Route, Switch } from "react-router-dom";

import PrivateRoute from "./PrivateRoute";

import Login from "./login";
import ServerList from "./server-list";
import NotFound from "./not-found";

const Routes = () => (
  <HashRouter basename="/">
    <div>
      <Switch>
        <Route path="/" exact component={Login} />
        <Route path="/server-list" exact component={PrivateRoute(ServerList)} />
        <Route component={NotFound} />
      </Switch>
    </div>
  </HashRouter>
);

export default Routes;
