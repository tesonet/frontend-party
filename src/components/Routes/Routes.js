import React from 'react';
import { Switch, Route } from 'react-router-dom';

import PrivateRoute from './PrivateRoute';
import App from '../App/App';
import Login from '../Login/Login';
import ServerList from '../ServerList/ServerList';
import PageNotFound from '../PageNotFound/PageNotFound';
import {
  ROUTE_SERVER_LIST,
  ROUTE_LOGIN,
} from '../../constants/routes';

export default () => (
  <App>
    <Switch>
      <PrivateRoute exact path={ROUTE_SERVER_LIST} component={ServerList} />
      <Route path={ROUTE_LOGIN} component={Login} />
      <Route component={PageNotFound} />
    </Switch>
  </App>
);
