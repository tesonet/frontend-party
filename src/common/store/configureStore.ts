import { applyMiddleware, compose, createStore } from 'redux';
import middleware from './middleware';
import reducer from './reducer';
import { IAppState } from './types';

const devTool =
  process.env.TARGET === 'browser' && window.__REDUX_DEVTOOLS_EXTENSION__
    ? window.__REDUX_DEVTOOLS_EXTENSION__()
    : <T>(f: T): T => f;

const configureStore = (preloadedState?: IAppState) => {
  const enhancedCreateStore = compose<typeof createStore>(
    applyMiddleware(...middleware),
    devTool
  )(createStore);

  return enhancedCreateStore(reducer, preloadedState as IAppState);
};

export default configureStore;
