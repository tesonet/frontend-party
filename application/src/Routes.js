import React, { Component } from 'react';
import { connect } from 'react-redux';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import App from './App';
import Servers from './components/servers/ServersPage';
import PageNotFound from './components/PageNotFound';
import { persistStore } from 'redux-persist';
import store from './Store';

class Routes extends Component {

  constructor() {
    super()
    this.state = { rehydrated: false }
  }

  componentWillMount(){
  persistStore(store, {}, () => {
    this.setState({ rehydrated: true })
  })
}

  render() {
    if(!this.state.rehydrated){
      return <h1>LOADING</h1>
    }
    return (
      <BrowserRouter>
        <Switch>
          <Route exact path='/' component={(this.props.login.token && this.props.login.userAuthenticated) ? Servers : App} />
          <Route path='*' component={PageNotFound} />
        </Switch>
      </BrowserRouter>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    login: state.LoginReducer
  }
}

export default connect(mapStateToProps)(Routes);
