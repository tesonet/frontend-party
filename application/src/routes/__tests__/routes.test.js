import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Routes from '../index.js';
import configureStore from 'redux-mock-store';

describe('Routes work as expected',() => {
  it('renders login page if user is not authenticated',() => {
    const state = {
      LoginReducer: {
        token: false,
        userAuthenticated: false
      }
    }

    const mockStore = configureStore();
    const store = mockStore(state);

    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <Routes />
        </MemoryRouter>
      </Provider>
    );

    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
    expect(component.find('.loginPage').length).toEqual(1);
    expect(component.find('.serversPage').length).toEqual(0);

  });

  it('renders servers page if user is authenticated',() => {
    const state = {
      LoginReducer: {
        token: true,
        userAuthenticated: true
      },
      ServersReducer: {
        errorMessage: false,
        servers: []
      }
    }

    const mockStore = configureStore();
    const store = mockStore(state);

    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <Routes />
        </MemoryRouter>
      </Provider>
    );

    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
    expect(component.find('.loginPage').length).toEqual(0);
    expect(component.find('.serversPage').length).toEqual(1);

  })
})
