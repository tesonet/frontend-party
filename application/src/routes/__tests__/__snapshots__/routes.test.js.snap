// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes work as expected renders login page if user is not authenticated 1`] = `
<div
  class="loginPage"
>
  <div
    class="container"
  >
    <img
      alt="Testio logo"
      class="logo"
      src="logo-white.png"
    />
    <div
      class="form"
    >
      <form>
        <div
          class="form-group"
        >
          <div
            class="input"
          >
            <svg
              class="username-icon"
              height="16"
              viewbox="0 0 14 16"
              width="14"
            >
              <path
                d="M582,383a4,4,0,0,0-4,4v1a4,4,0,0,0,8,0v-1A4,4,0,0,0,582,383Zm4.036,9.426a5.961,5.961,0,0,1-8.072,0A3.859,3.859,0,0,0,575,396v3h14v-3A3.859,3.859,0,0,0,586.036,392.426Z"
                id="ico-username"
                transform="translate(-575 -383)"
              />
            </svg>
            <input
              class="form-control"
              name="username"
              placeholder="Username"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="form-group"
        >
          <div
            class="input"
          >
            <svg
              class="password-icon"
              height="18"
              viewbox="0 0 14 18"
              width="14"
            >
              <path
                d="M582,458a4.012,4.012,0,0,0-4,4v3h-2a0.945,0.945,0,0,0-1,1v9a0.945,0.945,0,0,0,1,1h12a0.945,0.945,0,0,0,1-1v-9a0.945,0.945,0,0,0-1-1h-2v-3A4.012,4.012,0,0,0,582,458Zm2,7h-4v-3a2,2,0,0,1,4,0v3Zm-2,3a1,1,0,0,1,1,1v3a1,1,0,0,1-2,0v-3A1,1,0,0,1,582,468Z"
                id="ico-lock"
                transform="translate(-575 -458)"
              />
            </svg>
            <input
              class="form-control"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          class="btn"
          type="submit"
        >
          Log In
        </button>
      </form>
      <span />
    </div>
  </div>
</div>
`;

exports[`Routes work as expected renders servers page if user is authenticated 1`] = `
<div
  class="serversPage"
>
  <div
    class="header"
  >
    <div
      class="top row"
    >
      <div
        class="col-xs-6"
      >
        <img
          alt="Testio logo"
          src="logo-dark.png"
        />
      </div>
      <div
        class="col-xs-6 right log-out"
      >
        <div
          class="log-out-btn"
        >
          <svg
            class="logout-icon"
            height="16"
            viewbox="0 0 16 16"
            width="16"
          >
            <path
              d="M3027,54l-4,4,4,4V59h6V57h-6V54Zm11,12h-9a0.945,0.945,0,0,1-1-1V62h2v2h7V52h-7v2h-2V51a0.945,0.945,0,0,1,1-1h9a0.945,0.945,0,0,1,1,1V65A0.945,0.945,0,0,1,3038,66Z"
              id="ico-logout"
              transform="translate(-3023 -50)"
            />
          </svg>
          <span>
            Logout
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="info row"
  >
    <div
      class="col-xs-6"
    >
      Server
    </div>
    <div
      class="right col-xs-6"
    >
      Distance
    </div>
  </div>
  <div
    class="servers container-fluid"
  >
    <div
      style="position:relative;overflow:hidden;width:100%;height:100%;"
    >
      <div
        style="position:absolute;top:0;left:0;right:0;bottom:0;overflow:scroll;-webkit-overflow-scrolling:touch;margin-right:0;margin-bottom:0;"
      />
      <div
        style="position:absolute;height:6px;transition:opacity 200ms;opacity:0;display:none;right:2px;bottom:2px;left:2px;border-radius:3px;"
      >
        <div
          style="position:relative;display:block;height:100%;cursor:pointer;border-radius:inherit;background-color:rgba(0,0,0,.2);"
        />
      </div>
      <div
        style="position:absolute;width:6px;transition:opacity 200ms;opacity:0;display:none;right:2px;bottom:2px;top:2px;border-radius:3px;"
      >
        <div
          style="position:relative;display:block;width:100%;cursor:pointer;border-radius:inherit;background-color:rgba(0,0,0,.2);"
        />
      </div>
    </div>
  </div>
</div>
`;
